src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -43,154 +51,224 @@ export class QuickBar extends LitElement {
-                      <span slot="secondary" class="secondary"
-                        >${item.altText}</span
-                      >
-                    `
-                  : null}
-                ${this._commandTriggered === index
-                  ? html`<ha-circular-progress
-                      size="small"
-                      active
-                      slot="meta"
-                    ></ha-circular-progress>`
-                  : null}
-              </mwc-list-item>
-            `
-          )}
-        </mwc-list>
+          @focus=${this._setFocusFirstListItem}
+        >
+          ${this._commandMode
+            ? html`<ha-svg-icon
+                slot="prefix"
+                class="prefix"
+                .path=${mdiConsoleLine}
+              ></ha-svg-icon>`
+            : ""}
+        </search-input>
+        ${!this._items
+          ? html`<ha-circular-progress
+              size="small"
+              active
+            ></ha-circular-progress>`
+          : html`<mwc-list
+              activatable
+              @rangechange=${this._handleRangeChanged}
+              @keydown=${this._handleListItemKeyDown}
+              @selected=${this._handleSelected}
+              style=${styleMap({
+                height: `${Math.min(
+                  this._items.length * (this._commandMode ? 56 : 72) + 26,
+                  this._done ? 500 : 0
+                )}px`,
+              })}
+            >
+              ${scroll({
+                items: this._items,
+                renderItem: (item: QuickBarItem, index?: number) =>
+                  this._renderItem(item, index),
+              })}
+            </mwc-list>`}
       </ha-dialog>
     `;
   }
 
-  private async processItemAndCloseDialog(ev: SingleSelectedEvent) {
-    const index = ev.detail.index;
-    const item = (ev.target as any).items[index].item;
+  private _handleOpened() {
+    this._setFilteredItems();
+    this.updateComplete.then(() => {
+      this._done = true;
+    });
+  }
+
+  private async _handleRangeChanged(e) {
+    if (this._focusSet) {
+      return;
+    }
+    if (e.firstVisible > -1) {
+      this._focusSet = true;
+      await this.updateComplete;
+      this._setFocusFirstListItem();
+    }
+  }
+
+  private _renderItem(item: QuickBarItem, index?: number) {
+    return html`
+      <mwc-list-item
+        .twoline=${Boolean(item.altText)}
+        .item=${item}
+        index=${ifDefined(index)}
+        hasMeta
+        graphic=${item.altText ? "avatar" : "icon"}
+      >
+        <ha-icon .icon=${item.icon} slot="graphic"></ha-icon>
+        <span>${item.text}</span>
+        ${item.altText
+          ? html`
+              <span slot="secondary" class="secondary">${item.altText}</span>
+            `
+          : null}
+        ${this._commandTriggered === index
+          ? html`<ha-circular-progress
+              size="small"
+              active
+              slot="meta"
+            ></ha-circular-progress>`
```

Comment:
Moving the circular progress bar inside this async renderItem breaks the progress bar. We never go back into this function after `this._commandTriggered` gets a value so the progress bar never appears.